pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -472,3 +479,49 @@ func BuildDefaultHTTPRoute(clusterName string) *route.Route {
 		},
 	}
 }
+
+// translateFault translates networking.HTTPFaultInjection into Envoy's HTTPFault
+func translateFault(in *networking.HTTPFaultInjection) *xdshttpfault.HTTPFault {
+	if in == nil {
+		return nil
+	}
+
+	out := xdshttpfault.HTTPFault{}
+	if in.Delay != nil {
+		out.Delay = &xdsfault.FaultDelay{
+			Type:    xdsfault.FaultDelay_FIXED,
+			Percent: uint32(in.Delay.Percent),
+		}
+		switch d := in.Delay.HttpDelayType.(type) {
+		case *networking.HTTPFaultInjection_Delay_FixedDelay:
+			delayDuration := util.GogoDurationToDuration(d.FixedDelay)
+			out.Delay.FaultDelaySecifier = &xdsfault.FaultDelay_FixedDelay{
+				FixedDelay: &delayDuration,
+			}
+		default:
+			log.Warnf("Ignoring exponential fault in route")
```

Comment:
`Exponential faults are not yet supported in routes`

or similar; when we say "ignoring" it's not clear if its the user's fault (e.g. bad config) or that it's not supported.